flang/lib/Optimizer/Dialect/Support/FIRContext.cpp

```diff
@@ -114,6 +114,22 @@ mlir::LLVM::TargetFeaturesAttr fir::getTargetFeatures(mlir::ModuleOp mod) {
   return {};
 }
 
+void fir::setIdent(mlir::ModuleOp mod, llvm::StringRef ident) {
+  if (ident.empty())
+    return;
+
+  auto *ctx = mod.getContext();
```

Comment:
```suggestion
  mlir::MLIRContext *ctx = mod.getContext();
```
Nit: Only use auto when the type is given on the RHS or when it is too complex to write out.